app/models/user.rb

```diff
@@ -100,22 +100,22 @@ class User < ActiveRecord::Base
   scope :idle,
     :conditions => ["lastvisit < ?", 30.minutes.ago.utc]
 
-  validates_uniqueness_of :username, :email, :steamid
-  validates_length_of :firstname, :in => 1..15, :allow_blank => true
-  validates_length_of :lastname, :in => 1..25, :allow_blank => true
-  validates_length_of :username, :in => 2..20
-  validates_format_of :username, :with => /\A[A-Za-z0-9_\-\+]{2,20}\Z/
-    validates_presence_of :raw_password, :on => :create
-  validates_length_of :email, :maximum => 50
-  validates_format_of :email, :with => /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\Z/i
-  validates_length_of :steamid, :maximum => 30
-  validates_format_of :steamid, :with => /\A([0-9]{1,10}:){2}[0-9]{1,10}\Z/
-    validates_length_of :time_zone, :maximum => 100, :allow_blank => true, :allow_nil => true
-  validates_inclusion_of [:public_email], :in => [true, false], :allow_nil => true
+  before_validation :update_password
+
+  validates :username, uniqueness: true, length: {in: 2..20}, format: /\A[A-Za-z0-9_\-\+]{2,20}\Z/
+  validates :firstname, length: {in: 1..15}, allow_blank: true
+  validates :lastname, length: {in: 1..25}, allow_blank: true
+  validates :raw_password, presence: {on: :create}
+  validates :email, uniqueness: true, length: {maximum: 50}, format: /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\Z/i
```

Comment:
Line is too long. [114/80]<br>Space inside { missing.<br>Space inside } missing.